catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java

```diff
@@ -478,6 +512,13 @@ private String addColumnFieldDefinition(TableChange.AddColumn addColumn) {
         .append(dataType)
         .append(SPACE);
 
+    if (addColumn.isAutoIncrement()) {
+      Preconditions.checkArgument(
+          Types.allowAutoIncrement(addColumn.getDataType()),
```

Comment:
Can this simplified to `Types.allowAutoIncrement(addColumn.getDataType() || !addColumn.isAutoIncrement()`?